web-common/src/features/dashboards/stores/DashboardStateProvider.svelte

```diff
@@ -1,53 +1,27 @@
 <script lang="ts">
   import { page } from "$app/stores";
-  import {
-    createTimeRangeSummary,
-    useMetricsView,
-    useModelHasTimeSeries,
-  } from "@rilldata/web-common/features/dashboards/selectors/index";
   import { getStateManagers } from "@rilldata/web-common/features/dashboards/state-managers/state-managers";
-  import { metricsExplorerStore } from "@rilldata/web-common/features/dashboards/stores/dashboard-stores";
-  import { syncDashboardState } from "@rilldata/web-common/features/dashboards/stores/syncDashboardState";
+  import { createDashboardStateSync } from "@rilldata/web-common/features/dashboards/stores/syncDashboardState";
   import { initLocalUserPreferenceStore } from "@rilldata/web-common/features/dashboards/user-preferences";
-  import { createQueryServiceMetricsViewSchema } from "@rilldata/web-common/runtime-client";
-  import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+  import { writable } from "svelte/store";
   import Spinner from "../../entity-management/Spinner.svelte";
   import { EntityStatus } from "../../entity-management/types";
 
   export let metricViewName: string;
 
   $: initLocalUserPreferenceStore(metricViewName);
-  const stateManagers = getStateManagers();
-  const metricsView = useMetricsView(stateManagers);
-  const hasTimeSeries = useModelHasTimeSeries(stateManagers);
-  const timeRangeQuery = createTimeRangeSummary(stateManagers);
-  const metricsViewSchema = createQueryServiceMetricsViewSchema(
-    $runtime.instanceId,
-    metricViewName,
-  );
-
-  function syncDashboardStateLocal() {
-    syncDashboardState(
-      metricViewName,
-      $metricsView.data,
-      $metricsViewSchema.data?.schema,
-      $timeRangeQuery.data,
-      $page.url.searchParams.get("state"),
-    );
-  }
 
-  $: if (
-    $metricsView.data &&
-    $metricsViewSchema.data &&
-    ($timeRangeQuery.data || !$hasTimeSeries.data)
-  ) {
-    syncDashboardStateLocal();
-  }
-
-  $: ready = metricViewName in $metricsExplorerStore.entities;
+  const dashboardStateSync = createDashboardStateSync(
+    getStateManagers(),
+    writable({
+      isFetching: false,
+      error: "",
+      data: $page.url.searchParams.get("state") ?? "",
+    }),
```

Comment:
Feels a little hacky. Rather could this `data` object be computed via fallback/default logic inside the `createDashboardStateSync()` function?